tests/foreman/api/test_host.py

```diff
@@ -1438,3 +1438,45 @@ def test_positive_delete_and_check_host(self):
             host.read()
         except HTTPError:
             pytest.fail("HTTPError 404 raised unexpectedly!")
+
+
+class TestHostBulkAction:
+    """ Tests for host bulk actions. """
+
+    @pytest.mark.tier2
+    def test_positive_bulk_destroy(self, module_org):
+        """Destroy multiple hosts make sure that hosts were removed,
+        or were not removed when host is excluded from the list.
+
+        :id: 06d63376-8bf6-11eb-ab9f-98fa9b6ecd5a
+
+        :expectedresults: Included list of hosts,
+            that are not part of excluded list of host, are removed.
+
+        :CaseImportance: Medium
+        """
+
+        host_ids = []
+        for _ in range(3):
+            name = gen_choice(valid_hosts_list())
+            host = entities.Host(name=name, organization=module_org).create()
+            host_ids.append(host.id)
+
+        entities.Host().bulk_destroy(
+            data={
+                'organization_id': module_org.id,
+                'included': {'ids': host_ids},
+                'excluded': {'ids': host_ids[:-1]},
+            }
+        )
+        for host_id in host_ids[:-1]:
+            entities.Host(id=host_id).read()
```

Comment:
maybe assert for result return code, or for something in the result content

Reply:
This action will pass only if the hosts were excluded from the list.  I did not care about the read result. 

Please write me more, what would you expect to be there and why ?

Reply:
without some kind of check for the read contents or read success, what is the purpose of this loop?

Reply:
the read checks whether the excluded hosts were really excluded, 
read would fail if not

Reply:
Without the read raising an exception, the test wouldn't know that the read failed.

Reply:
@tstrych we want to make these tests future proof, so some additional asserts could help us to catch issues that could go unnoticed otherwise (like some different code returned, or read passing successfully but returning wrong host), something like this (not guaranteeing these are the exact api calls):
```
result = entities.Host(id=host_id).read()
assert result.status == 0
assert result.stdout.id == host_id
```

Reply:
you could remove the status check in the above example and rely solely on checking the host id

Reply:
thank you for  your explanation :+1: